Zilon.Core/Zilon.Core/World/BiomeInitializer.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Zilon.Core.MapGenerators;
+using Zilon.Core.Schemes;
+
+namespace Zilon.Core.World
+{
+    public class BiomeInitializer : IBiomeInitializer
+    {
+        private readonly ISectorGenerator _sectorGenerator;
+        private readonly IBiomeSchemeRoller _biomeSchemeRoller;
+
+        public BiomeInitializer(ISectorGenerator sectorGenerator, IBiomeSchemeRoller biomeSchemeRoller)
+        {
+            _sectorGenerator = sectorGenerator ?? throw new ArgumentNullException(nameof(sectorGenerator));
+            _biomeSchemeRoller = biomeSchemeRoller ?? throw new ArgumentNullException(nameof(biomeSchemeRoller));
+        }
+
+        public async Task<IBiome> InitBiomeAsync(ILocationScheme locationScheme)
+        {
+            var biom = new Biome(locationScheme);
+
+            await CreateStartSectorAsync(biom).ConfigureAwait(false);
+
+            return biom;
+        }
+
+        public async Task MaterializeLevel(ISectorNode sectorNode)
+        {
+            if (sectorNode is null)
+            {
+                throw new ArgumentNullException(nameof(sectorNode));
+            }
+
+            if (sectorNode.State != SectorNodeState.SchemeKnown)
+            {
+                throw new InvalidOperationException();
+            }
+
+            var biom = sectorNode.Biome;
+
+            // Важно генерировать соседние узлы до начала генерации сектора,
+            // чтобы знать переходы из сектора.
+
+            await CreateNextSectorNodesAsync(sectorNode, biom).ConfigureAwait(false);
+
+            var sector = await _sectorGenerator.GenerateAsync(sectorNode).ConfigureAwait(false);
+
+            sectorNode.MaterializeSector(sector);
+        }
+
+        private async Task<SectorNode> RollAndBindBiomeAsync()
+        {
+            var rolledLocationScheme = _biomeSchemeRoller.Roll();
+
+            var biome = new Biome(rolledLocationScheme);
+
+            var startSectorScheme = biome.LocationScheme.SectorLevels.Single(x => x.IsStart);
+
+            var newBiomeSector = new SectorNode(biome, startSectorScheme);
+
+            var sector = await _sectorGenerator.GenerateAsync(newBiomeSector).ConfigureAwait(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'sector'.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5260137)